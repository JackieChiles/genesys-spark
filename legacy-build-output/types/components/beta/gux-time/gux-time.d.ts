import { JSX } from '../../../stencil-public-runtime';
import { GuxDateTimeFormat } from '../../../i18n/DateTimeFormatter';
export declare class GuxTime {
  private formatter;
  /**
   * Reference to the host element.
   */
  root: HTMLElement;
  /**
   * The ISO string representation of the time to format
   */
  datetime: string;
  /**
   * Format option type
   */
  format: GuxDateTimeFormat;
  componentWillLoad(): void;
  render(): JSX.Element;
}
