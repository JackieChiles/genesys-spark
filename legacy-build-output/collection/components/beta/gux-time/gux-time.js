import { h, Host } from '@stencil/core';
import { trackComponent } from '@utils/tracking/usage';
import { DateTimeFormatter } from '../../../i18n/DateTimeFormatter';
import { getDesiredLocale } from '../../../i18n/index';
export class GuxTime {
  constructor() {
    this.datetime = new Date().toISOString();
    this.format = 'short';
  }
  componentWillLoad() {
    trackComponent(this.root);
    this.formatter = new DateTimeFormatter(getDesiredLocale(this.root));
  }
  render() {
    return (h(Host, null, this.formatter.formatTime(new Date(this.datetime), this.format)));
  }
  static get is() { return "gux-time-beta"; }
  static get encapsulation() { return "shadow"; }
  static get properties() {
    return {
      "datetime": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "The ISO string representation of the time to format"
        },
        "attribute": "datetime",
        "reflect": false,
        "defaultValue": "new Date().toISOString()"
      },
      "format": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "GuxDateTimeFormat",
          "resolved": "\"full\" | \"long\" | \"medium\" | \"short\"",
          "references": {
            "GuxDateTimeFormat": {
              "location": "import",
              "path": "../../../i18n/DateTimeFormatter"
            }
          }
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "Format option type"
        },
        "attribute": "format",
        "reflect": false,
        "defaultValue": "'short'"
      }
    };
  }
  static get elementRef() { return "root"; }
}
