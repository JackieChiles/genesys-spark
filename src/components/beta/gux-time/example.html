<h1>gux-time-beta</h1>

<h2>Times</h2>

<h3>Default</h3>
<gux-time-beta></gux-time-beta>

<h3>Short</h3>
<gux-time-beta id="time-1" format="short"></gux-time-beta>

<h3>Medium</h3>
<gux-time-beta id="time-2" format="medium"></gux-time-beta>

<h3>Long</h3>
<gux-time-beta id="time-3" format="long"></gux-time-beta>

<h3>Full</h3>
<gux-time-beta id="time-4" format="full"></gux-time-beta>

<style
  onload="(function() {
    const time1 = document.querySelector('#time-1');
    const time2 = document.querySelector('#time-2');
    const time3 = document.querySelector('#time-3');
    const time4 = document.querySelector('#time-4');

    const date = new Date().toISOString();

    time1.time = date;
    time2.time = date;
    time3.time = date;
    time4.time = date;
  })()"
>
  .not-used {
    -custom-noop: noop;
  }
</style>
