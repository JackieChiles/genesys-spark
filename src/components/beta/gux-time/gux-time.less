@import (reference) '../../../style/spacing.less';

.gux-time {
    display: flex;
    padding: @spacing-xs @spacing-xs;
}