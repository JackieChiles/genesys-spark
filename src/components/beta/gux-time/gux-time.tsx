import { Component, h, Prop, Element } from '@stencil/core';
import { trackComponent } from '../../../usage-tracking';
import { DateTimeFormatter, GuxDateTimeFormat } from '../../../i18n/DateTimeFormatter';
import { getDesiredLocale } from '../../../i18n/index';

@Component({
  styleUrl: 'gux-time.less',
  tag: 'gux-time-beta',
  shadow: true
})
export class GuxTime {
  private formatter : DateTimeFormatter;

  /**
   * Reference to the host element.
   */
  @Element()
  root: HTMLElement;

  /**
   * The Date object to format 
   */
  @Prop()
  date: Date = new Date();

  /**
   * Format option type 
   */
  @Prop()
  format: GuxDateTimeFormat = 'full';

  componentWillLoad(): void {
    trackComponent(this.root);
    this.formatter = new DateTimeFormatter(getDesiredLocale(this.root));
  }

  render() {
    return (<span>{ this.formatter.formatTime(this.date, this.format) }</span>);
  }
}
