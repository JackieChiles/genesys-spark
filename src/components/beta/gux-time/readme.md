# gux-time-beta



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description                                         | Type                                      | Default                    |
| -------- | --------- | --------------------------------------------------- | ----------------------------------------- | -------------------------- |
| `time`   | `time`    | The ISO string representation of the time to format | `string`                                  | `new Date().toISOString()` |
| `format` | `format`  | Format option type                                  | `"full" \| "long" \| "medium" \| "short"` | `'full'`                   |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
