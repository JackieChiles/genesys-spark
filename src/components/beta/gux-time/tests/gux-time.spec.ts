import { newSpecPage } from '@stencil/core/testing';
import { GuxTime } from '../gux-time';

const components = [GuxTime];
const language = 'en';
const date = new Date(2022, 6, 7, 9, 35, 30, 100).toISOString();

describe('#render different time formats eg(short,medium,full,long)', () => {
  [
    `<gux-time-beta datetime=${date} format="short"></gux-time-beta>`,
    `<gux-time-beta datetime=${date} format="medium"></gux-time-beta>`,
    `<gux-time-beta datetime=${date} format="full"></gux-time-beta>`,
    `<gux-time-beta datetime=${date} format="long"></gux-time-beta>`
  ].forEach((html, index) => {
    it(`should render component as expected (${index + 1})`, async () => {
      const page = await newSpecPage({ components, html, language });

      expect(page.root).toMatchSnapshot();
    });
  });
});
