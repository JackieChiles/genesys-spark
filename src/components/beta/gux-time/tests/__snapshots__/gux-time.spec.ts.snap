// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render different time formats eg(short,medium,full,long) should render component as expected (1) 1`] = `
<gux-time-beta datetime="2022-07-07T13:35:30.100Z" format="short">
  <mock:shadow-root>
    9:35 AM
  </mock:shadow-root>
</gux-time-beta>
`;

exports[`#render different time formats eg(short,medium,full,long) should render component as expected (2) 1`] = `
<gux-time-beta datetime="2022-07-07T13:35:30.100Z" format="medium">
  <mock:shadow-root>
    9:35:30 AM
  </mock:shadow-root>
</gux-time-beta>
`;

exports[`#render different time formats eg(short,medium,full,long) should render component as expected (3) 1`] = `
<gux-time-beta datetime="2022-07-07T13:35:30.100Z" format="full">
  <mock:shadow-root>
    9:35:30 AM GMT-04:00
  </mock:shadow-root>
</gux-time-beta>
`;

exports[`#render different time formats eg(short,medium,full,long) should render component as expected (4) 1`] = `
<gux-time-beta datetime="2022-07-07T13:35:30.100Z" format="long">
  <mock:shadow-root>
    9:35:30 AM GMT-4
  </mock:shadow-root>
</gux-time-beta>
`;
